wrangler/src/trust/trust.js

```diff
@@ -0,0 +1,271 @@
+}
+
+/**
+ * Given a public key, return the home domain of the account
+ * @param pubkey - The public key of the account you want to get the home domain of.
+ * @returns The home domain of the account.
+ */
+export async function getHomeDomain(pubkey) {
+    const turretaccountinfo = getAccountInfo(pubkey)
+    if (turretaccountinfo.home_domain){
+        return turretaccountinfo.home_domain
+    } return undefined
+}
+/**
+ * It gets the signer for the TxFunction for the local turret.
+ * @param txfunctionhash - The hash of the TxFunction you want the signer for.
+ * @returns The function signer's public key.
+ */
+export async function getLocalFunctionSigner(txfunctionhash){
+    const { value, metadata } = await TX_FUNCTIONS.getWithMetadata(txfunctionhash, 'arrayBuffer')
+    if (!value)
+        throw {status: 404, message: `txFunction could not be found this turret`}
+    const { txFunctionSignerPublicKey } = metadata
+    return txFunctionSignerPublicKey
+}
+
+/**
+ * It returns the high threshold for the account.
+ * @param pubkey - The public key of the account you want to get the threshold for.
+ * @returns the high_threshold value.
+ */
+export async function getAccountThreshold(pubkey){
+    const sourceaccountinfo = await getAccountInfo(pubkey)
+    const requiredthreshold = sourceaccountinfo.thresholds.high_threshold
+    if (requiredthreshold)
+        return requiredthreshold
+    throw ('unable to obtain threshold information')
+}
+
+/**
+ * It returns the weight of a signer for a given account.
+ * @param signer - The public key of the signer you want to get the weight of
+ * @param account - the account to get the signer weight for
+ * @returns The weight of the signer.
+ */
+export async function getSignerWeight(signer, account){
+    const accountinfo = await getAccountInfo(account)
+    const accountsigners = accountinfo.signers
+    for (const signer in accountsigners){
+        if (accountsigners[signer].key.includes(signer))
+            return accountsigners[signer].weight
+    }
+    throw (`the signer ${signer} is not a signer for account ${account}`)
+}
+/**
+ * Get the sequence number for a given public key
+ * @param pubkey - The public key of the account you want to get the sequence number for.
+ * @returns The sequence number of the account.
+ */
+export async function getSequenceNumber(pubkey){
+    accountinfo = getAccountInfo(pubkey)
+    return accountinfo.sequence
+}
+/**
+ * This function checks the local toml to see if the turret address is trusted
+ * @param turret - the address of the turret you want to check for trust
+ * @returns A boolean value.
+ */
+export async function checkLocalQuorem(turret) {
+    const localquorem = localtoml()
+    try{
+        const thetoml = parse(localquorem).TSS.TURRETS
+        const quorem = []
+        for (const each in thetoml){
+            quorem.push(thetoml[each])
+        }
+        if(quorem.includes(turret))
+            return true
+        return false
```

Comment:
was the TOML parsing going to be for the new TOML format we discussed?

It appears this is still the old format

Reply:
Let's do the new format.